### PR TITLE
zos: implement uv__io_check_fd

### PR DESCRIPTION
**Pull request by jBarz**
This method uses the poll syscall to determine whether POLLNVAL is
flagged or not.

### CODE REVIEW

**Comment by saghul** [comment]
Hey @jBarz sorry for the delay. Please keep the PRs comming. I'll review them in order as soon as I can.

File: src/unix/os390.c [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+ * IN THE SOFTWARE.
+ */
+
+#include "internal.h"
+
+int uv__io_check_fd(uv_loop_t* loop, int fd) {
+
```
Comment (bnoordhuis) [inline src/unix/os390.c]: Tiny style nit: can you drop the blank line?

File: src/unix/os390.c [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  struct pollfd p[1];
+  p[0].fd = fd;
+  p[0].events = 0 | POLLIN;
```
Comment (bnoordhuis) [inline src/unix/os390.c]: Out of curiosity, why do you OR in (a no-op) zero?

File: src/unix/os390.c [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+  if (poll(p, 1, 0) == -1)
```
Comment (bnoordhuis) [inline src/unix/os390.c]: Do you need to handle EINTR here?

File: src/unix/os390.c (hunk 2 above)
Comment (jBarz) [inline src/unix/os390.c]: No reason. We don't need that so I will instead do the following.
p[0].events = POLLIN

File: src/unix/os390.c (hunk 3 above)
Comment (jBarz) [inline src/unix/os390.c]: Indeed we do. I will poll while (rv == -1 && errno == EINTR)

**Comment by saghul** [comment]
LGTM. Can you rebase this?

**Comment by saghul** [comment]
After merging other PRs for z/OS this one no longer applies cleanly, can rebase one last time?

**Comment by jBarz** [comment]
Done

**Comment by saghul** [comment]
Landed in 9e641d2, thanks!

**Comment by saghul** [comment]
Quick question (I might have asked this before, sorry!): does the build with GYP actually work in z/OS?

**Comment by jBarz** [comment]
Yes, GYP works!
We are only using gyp in our builds

**Comment by saghul** [comment]
Okidoki! 👍

**Pull request closed without merging**